include/envoy/server/filter_config.h

```diff
@@ -181,11 +181,35 @@ class NamedListenerFilterConfigFactory {
   virtual std::string name() PURE;
 };
 
+class ProtocolOptionsConsumer {
```

Comment:
Perhaps this should be called ProtocolOptionsFactory? I'm sure what "consumer" means here.

Regardless, should add at least a brief class comment here.

Reply:
Sounds good to me - will fix

Reply:
what about this comment for class doc:
`Implemented by filter factories that require more options to process the protocol used by the upstream cluster.`